packages/block-library/src/navigation-link/index.php

```diff
@@ -226,17 +226,56 @@ function render_block_core_navigation_link( $attributes, $content, $block ) {
 	return $html;
 }
 
+/**
+ * Returns a navigation link variation
+ *
+ * @param WP_Taxonomy|WP_Post_Type $entity post type or taxonomy entity.
+ * @param string                   $kind string of value 'taxonomy' or 'post-type'.
+ *
+ * @return array
+ */
+function build_variation_for_navigation_link( $entity, $kind ) {
+	$name = 'post_tag' === $entity->name ? 'tag' : $entity->name;
+	return array(
+		'name'        => $name,
+		/* translators: %s: Entity type, eg: Post Link, Page Link, Category Link, Tag Link, Portfolio Link etc */
+		'title'       => sprintf( __( '%s Link' ), $entity->labels->singular_name ),
```

Comment:
@gziolo any recommendations on who to ask for i18n advice here? It's not ideal that we're using another translated label to build up this string.

Reply:
I think it's fine and follows best practices as listed at https://codex.wordpress.org/I18n_for_WordPress_Developers.

@swissspidy, can you confirm if it is the right way to go?

Reply:
`get_post_type_labels()` shoulld be extended in core to have a new `item_link` entry set to `Post Link` / `Page Link`. Same for `A link to a %s.` below (maybe with key `item_link_description`?).

For example, in German the translation for `A link to a %s.` changes depending on the post type.

`A link to a post` -> `Ein Link zu einem Beitrag`
`A link to a page` -> `Ein Link zu einer Seite`

`sprintf( __( 'A link to a %s.' ), $entity->labels->singular_name )` doesn't allow for that.


Related: https://core.trac.wordpress.org/ticket/51387 and https://github.com/WordPress/gutenberg/issues/19144#issuecomment-698212378

Reply:
Thank you @swissspidy. I can see how speaking German makes it easier to reason about those challenges with translations. In Polish it would be:
- `Link do wpisu`
- `Link do strony`

When I think more about it, it gets even more fun in Polish because you couldn't share `$entity->labels->singular_name`:
- `To jest wpis` (This is post)
- `To jest strona` (This is page)

So I guess your proposal should address those type of issues.

Reply:
Thanks for the guidance here @swissspidy @gziolo. I can open a trac issue and create a patch for a new entry in `get_post_type_labels`

Reply:
I prototyped the change in https://github.com/WordPress/wordpress-develop/pull/1015/commits/298bbd1007128fb4def74948e5b1ab774c3f10fa and https://github.com/WordPress/gutenberg/pull/29095/commits/5df7ec4fee9109b8e793203fea15e46d0fd97120. Happy to pull this out into a smaller PR.

I ended up adding some fallback logic, so we wouldn't end up with "Post Link" labels for other CPTs that haven't defined it.

| New Label | Fallback (notice the capitalization in description because this is built with singular_name) |
|-----|-----|
| <img width="672" alt="Screen Shot 2021-02-24 at 1 38 08 PM" src="https://user-images.githubusercontent.com/1270189/109070797-cf08ab80-76a7-11eb-94c1-3d3124cc7b0f.png"> | <img width="710" alt="Screen Shot 2021-02-24 at 1 38 01 PM" src="https://user-images.githubusercontent.com/1270189/109070715-b1d3dd00-76a7-11eb-8c0a-dd32b51b1fe5.png"> |

Reply:
Decided to pull out the fallback so behavior looks like this if labels are undefined:
<img width="1630" alt="Screen Shot 2021-03-01 at 8 58 26 AM" src="https://user-images.githubusercontent.com/1270189/109536326-730ca100-7a72-11eb-8c82-24e833df50d9.png">